server/users/users.controller.ts

```diff
@@ -68,6 +71,16 @@ export class UsersController {
         }
     }
 
+    @Get("admin")
+    @UseGuards(AbilitiesGuard)
+    @CheckAbilities(new AdminUserAbility())
+    public async admin(@Req() req: Request, @Res() res: Response) {
+        const parsedUrl = parse(req.url, true);
+        await this.viewService
+            .getNextServer()
+            .render(req, res, "/admin-page", Object.assign(parsedUrl.query));
+    }
+
     @Put("api/user/:id/password")
```

Comment:
Since you're here: it just come to my mind that we shouldn't be allowing the user id in any parameter for password changes, the user id should always come from the session token.